packages/shared/src/effect.ts

```diff
@@ -24,72 +24,72 @@ export class FetchContext extends Context.Tag("uploadthing/FetchContext")<
   FetchContextService
 >() {}
 
-interface ResponseWithBoundRequest extends ResponseEsque {
-  request: Request;
+interface ResponseWithURL extends ResponseEsque {
+  requestUrl: string;
 }
 
 // Temporary Effect wrappers below.
 // TODO should be refactored with much love
 // TODO handle error properly
 export const fetchEff = (
-  input: RequestInfo | URL,
+  input: string | URL,
   init?: RequestInit,
-): Effect.Effect<ResponseWithBoundRequest, FetchError, FetchContext> =>
+): Effect.Effect<ResponseWithURL, FetchError, FetchContext> =>
   Effect.flatMap(FetchContext, ({ fetch, baseHeaders }) => {
-    const req = new Request(input, {
-      ...init,
+    const reqInfo = {
+      url: input.toString(),
+      method: init?.method,
+      body: init?.body,
```

Comment:
Yeah thats a solution too, I fell in love with `Request` it seems haha

Reply:
HappyDOM, MSW and the Request constructor doesn't play very nice together... and also the logs should be a bit cleaner if we only store what we really care about and not all the other fuzz